src/service/searchService.ts

```diff
@@ -33,7 +33,7 @@ export class SearchService {
         } else {
             //Fix the results which are stored as string
             results.forEach(result => {
-                result.pubDate = new Date((result.pubDate as unknown as string));
+                result.pubDate = result.pubDate ? new Date((result.pubDate as unknown as string)) : undefined;
```

Comment:
Some shows (Red Dwarf after season 8, for example) were never broadcast on BBC, so the date is undefined. This results in a bug (currently in live) where you find a result, tap it, go back and it has disappeared from the now-loaded-from-cache list.